flepimop/gempyor_pkg/src/gempyor/NPI/SinglePeriodModifier.py

```diff
@@ -115,6 +115,8 @@ def __createFromConfig(self, npi_config):
         self.affected_subpops = set(self.subpops)
         if npi_config["subpop"].exists() and npi_config["subpop"].get() != "all":
             self.affected_subpops = {str(n.get()) for n in npi_config["subpop"]}
+            # subsamples the subpopulations to only the ones specified in the geodata (self.subpops)
+            self.affected_subpops = list(set(self.affected_subpops).intersection(self.subpops))
```

Comment:
two notes:
 - this is different implementation from MultiPeriodModifier
 - but seems to be the same thing; is there potential for an inheritance hierarchy here, so we only write this once?

Reply:
(this refactor is probably its own issue / PR)

Reply:
yes, agree, overtly complex